src/WebJobs.Script.WebHost/ConcurrencyOptionProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Script.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class ConcurrencyOptionProvider : IConcurrencyOptionProvider
```

Comment:
Not sure why this is needed. If our goal is to simply include ConcurrencyOptions in the payload, FunctionSyncManager can simply take the options as a ctor param as normal

Reply:
We are having `IServiceCollection` and `IServiceProvider` inside of the `ExtensionsOptionProvider` we can't get it from `FunctionsSyncManager`. If I add it to `ExtensionsOptionProvider` it might be over responsibility because of the Naming. One idea is, combine ExtensionsOptionOpvider and ConcurrencyOptionProvider with `SyncTriggerOptionProvider` then it might be Ok. I'll explain why I don't use the DI hydration, I'll explain it the following comment.

Reply:
Removed